src/wallet/spend.cpp

```diff
@@ -778,6 +778,10 @@ bool CWallet::CreateTransactionInternal(
         fee_needed = coin_selection_params.m_effective_feerate.GetFee(nBytes);
     }
 
+    // The only time that fee_needed should be less than the amount available for fees (in change_and_fee - change_amount) is when
+    // we are subtracting the fee from the outputs. If this occurs at any other time, it is a bug.
+    assert(coin_selection_params.m_subtract_fee_outputs || fee_needed <= change_and_fee - change_amount);
+
     // Update nFeeRet in case fee_needed changed due to dropping the change output
     if (fee_needed <= change_and_fee - change_amount) {
         nFeeRet = change_and_fee - change_amount;
```

Comment:
nit: this logic looks a little bit funny now. I think it might make more sense to have modified this `if` block as
```
￼    if (fee_needed <= change_and_fee - change_amount) {
￼        nFeeRet = change_and_fee - change_amount;
    } else {
    ￼    // The only time that fee_needed should be less than the amount available for fees (in change_and_fee - change_amount) is when
    ￼    // we are subtracting the fee from the outputs. If this occurs at any other time, it is a bug.
￼        assert(coin_selection_params.m_subtract_fee_outputs);
    }
```
which is logically equivalent but doesn't do the same check twice

Reply:
JFYI #25647 changes these lines